cert/cert.go

```diff
@@ -223,27 +335,61 @@ func UnmarshalEd25519PublicKey(b []byte) (ed25519.PublicKey, []byte, error) {
 }
 
 // Sign signs a nebula cert with the provided private key
-func (nc *NebulaCertificate) Sign(key ed25519.PrivateKey) error {
+func (nc *NebulaCertificate) Sign(key []byte) error {
```

Comment:
It would be nice if we knew the curve of the private key here as well. The underlying sign is catching both cases though.

P256 used to sign a X25519 error: `input overflows the modulus`
Ed25519 used to sign a P256 error: `panic: ed25519: bad private key length: 32`

Reply:
I think I'll do a less invasive change of adding the curve to the method signature, but I would like to add a PublicKey and PrivateKey struct with the Curve next to the key to make things simpler in a future cleanup PR

Reply:
added with [b4bec48](https://github.com/slackhq/nebula/pull/769/commits/b4bec4834133dfd6899596014c20bb1fc64ae81e)